aspnetcore/migration/1x-to-2x/index.md

```diff
@@ -69,36 +69,54 @@ In the *.csproj* file, update the `Version` attribute of each `<DotNetCliToolRef
 
 Rename both the node and variable to `AssetTargetFallback`:
 
-[!code-xml[Main](../1x-to-2x/samples/AspNetCoreDotNetCore2.0App/AspNetCoreDotNetCore2.0App/AspNetCoreDotNetCore2.0App.csproj?range=5)]
+[!code-xml[Main](../1x-to-2x/samples/AspNetCoreDotNetCore2App/AspNetCoreDotNetCore2App/AspNetCoreDotNetCore2App.csproj?range=4)]
 
 <a name="program-cs"></a>
 
 ## Update Main method in Program.cs
 In 1.x projects, the `Main` method of *Program.cs* looked like this:
 
-[!code-csharp[Main](../1x-to-2x/samples/AspNetCoreDotNetCore1.1App/AspNetCoreDotNetCore1.1App/Program.cs?name=snippet_ProgramCs&highlight=8-19)]
+[!code-csharp[Main](../1x-to-2x/samples/AspNetCoreDotNetCore1App/AspNetCoreDotNetCore1App/Program.cs?name=snippet_ProgramCs&highlight=8-19)]
 
 In 2.0 projects, the `Main` method of *Program.cs* has been simplified:
 
-[!code-csharp[Main](../1x-to-2x/samples/AspNetCoreDotNetCore2.0App/AspNetCoreDotNetCore2.0App/Program.cs?highlight=8-11)]
+[!code-csharp[Main](../1x-to-2x/samples/AspNetCoreDotNetCore2App/AspNetCoreDotNetCore2App/Program.cs?highlight=8-11)]
 
-The adoption of this new 2.0 pattern is highly recommended and is required for product features like [Entity Framework Core Migrations](xref:data/ef-mvc/migrations) to work. For example, running `Update-Database` from the Package Manager Console window or `dotnet ef database update` from the command line (on projects converted to ASP.NET Core 2.0) generates the following error:
+The adoption of this new 2.0 pattern is highly recommended and is required for product features like [Entity Framework (EF) Core Migrations](xref:data/ef-mvc/migrations) to work. For example, running `Update-Database` from the Package Manager Console window or `dotnet ef database update` from the command line (on projects converted to ASP.NET Core 2.0) generates the following error:
 
 ```
 Unable to create an object of type '<Context>'. Add an implementation of 'IDesignTimeDbContextFactory<Context>' to the project, or see https://go.microsoft.com/fwlink/?linkid=851728 for additional patterns supported at design time.
 ```
 
+<a name="db-init-code"></a>
+
+## Move database initialization code
+In 1.x projects using EF Core 1.x, a command such as `dotnet ef migrations add` does the following:
+1. Instantiates a `Startup` instance
+2. Invokes the `ConfigureServices` method to register all services with dependency injection (including `DbContext` types)
+3. Performs its requisite tasks
+
+In 2.0 projects using EF Core 2.0, the steps outlined above still occur. Additionally, the `Configure` method is invoked after `ConfigureServices`. If your 1.x app invoked database initialization code in its `Configure` method, unexpected problems can occur. For example, if the database doesn't yet exist, the seeding code will run before the EF Core Migrations command execution. This problem causes a `dotnet ef migrations list` command to fail if the database doesn't yet exist.
```

Comment:
ConfigureServices is not run in 2.0. /cc @bricelam for more detailed review on this.

Reply:
I think the cleanest way to think about this is that it is bad to do anything in BuildWebHost except to build and configure the web host. Anything that is about running the application should go outside of BuildWebHost--usually in Main. This allows tooling to get a WebHost without having side-effects related to starting the application.

Reply:
> In 2.0 projects using EF Core 2.0, the steps outlined above still occur. Additionally, the `Configure` method is invoked after `ConfigureServices`.

More like: In 2.0 projects using EF Core 2.0, `Program.BuildWebHost` is invoked to obtain the application services. Unlike 1.x, this has the additional side affect of invoking `Startup.Configure`.

(But I'm also OK over-simplifying if it makes it easier to understand.)

Reply:
I might also explicitly call out that the semantics of `Startup.Configure` are to configure the request pipeline, so using it as an app initialization hook is not correct.